source/upgrade/upgrade-mattermost-kubernetes-ha.rst

```diff
@@ -0,0 +1,232 @@
+
+  .. code-block:: bash
+
+    kubectl get pods -n mattermost
+    kubectl logs -f <pod-name> -n mattermost
+
+New pods are started with the updated version. Old pods are gracefully terminated after health checks pass, and traffic remains uninterrupted due to rolling upgrade behavior. NGINX, HAProxy, or Ingress configurations continue routing traffic seamlessly during the upgrade process.
+
+.. _active-active-upgrade-considerations:
+
+Step 3: Active/Active upgrade considerations
+--------------------------------------------
+
+When your deployment includes an Active/Active configuration with multi-cluster or multi-region deployments, you must ensure that all clusters are upgraded in a coordinated manner to maintain data consistency and service availability.
+
+For coordinated site upgrades, we recommend the following steps:
+
+- Use your global load balancer or DNS to route traffic away from the site being upgraded.
+- Upgrade one site at a time while directing traffic to the other. This minimizes downtime and allows for validation of the upgrade process.
+- Confirm replication health for databases and storage before proceeding to the next site.
+- Monitor performance and logs before resuming traffic to the upgraded site.
+
+To prevent split-brain scenarios during upgrades and ensure data consistency:
+
+- Use a global load balancer or DNS to direct traffic exclusively to the active site.
+- Disable writes on the site being upgraded until the upgrade is complete and validated.
+- Confirm that only one site is actively writing data at any time.
+- Monitor logs for signs of replication lag, file synchronization issues, or data divergence.
+- Validate the health of data replication channels before and after each site upgrade.
+
+Step 4: Validate
+------------------
+
+After the upgrade:
+
+1. Confirm all pods are running the new version using the following command:
+
+  .. code-block:: bash
+
+    kubectl get pods -n mattermost -o=jsonpath='{.items[*].spec.containers[*].image}'
+
+   Ensure that all pods are healthy and running the expected version.
+
+3. Perform product smoke tests:
+
+  - Log in to the Mattermost web interface and navigate across teams and channels.
+  - Test core functionalities such as posting messages and uploading files.
+  - Verify that all integrations, webhooks, and plugins are functioning as expected.
+  - Check for error messages in the logs.
+
+4. Use monitoring tools to confirm application health and performance. See the :doc:`Performance monitoring with Prometheus and Grafana </scale/deploy-prometheus-grafana-for-performance-monitoring>` documentation or the :doc:`Metrics plugin </scale/performance-monitoring-metrics>` documentation for details on collecting and reviewing performance metrics.
+
+Rollback strategy
+------------------
+
+In case of upgrade issues, use Helm’s rollback command:
+
+.. code-block:: bash
+
+    helm rollback mattermost <revision_number>
```

Comment:
Rollback should be made by modifying the custom resource and setting the Mattermost version to it's original value.